src/EventStore.Core/TransactionLog/FileNamingStrategy/VersionedPatternFileNamingStrategy.cs

```diff
@@ -44,6 +45,73 @@ public class VersionedPatternFileNamingStrategy : IFileNamingStrategy {
 			return versions;
 		}
 
+		private int GetFileNumberFor(string path) {
+			var fileName = Path.GetFileName(path);
+			if (!_chunkNamePattern.IsMatch(fileName))
+				throw new ArgumentException($"Invalid file name: {fileName}");
+
+			var start = _prefix.Length;
+			var end = fileName.IndexOf('.', _prefix.Length);
+			Debug.Assert(end != -1);
+
+			if (!int.TryParse(fileName[start..end], out var chunkNumber))
+				throw new ArgumentException($"Invalid file name: {fileName}");
+
+			return chunkNumber;
+		}
+
+		private int GetFileVersionFor(string path) {
+			var fileName = Path.GetFileName(path);
+			if (!_chunkNamePattern.IsMatch(fileName))
+				throw new ArgumentException($"Invalid file name: {fileName}");
+
+			var dot = fileName.IndexOf('.', _prefix.Length);
+			Debug.Assert(dot != -1);
+
+			if (!int.TryParse(fileName[(dot+1)..], out var chunkVersion))
+				throw new ArgumentException($"Invalid file name: {fileName}");
+
+			return chunkVersion;
+		}
+
+		public void EnumerateAllFiles(
+			Func<string, int, int, int> getNextFileNumber,
```

Comment:
The logic in here makes this class chunk specific, but it's named generically still.

Recommend taking this method (and the couple of private methods it uses) out to another, chunk specific class that takes a IFileNamingStrategy as a parameter to the constructor. Then it can also enumerate the files into an array during construction and resuse the array for each call to EnumerateAllFiles. On a typical db opening that would reduce the number of EnumerateFiles calls from 3 to 1. (and 2 to 1 in the truncator)

Reply:
Resolved in fb6fc57ee056288bd54f167649fc637a01b876db